src/Compilers/Core/Portable/Binding/BindingDiagnosticBag.cs

```diff
@@ -177,18 +190,20 @@ internal void AddDependencies(UseSiteInfo<TAssemblySymbol> useSiteInfo)
 
         internal void AddDependencies(CompoundUseSiteInfo<TAssemblySymbol> useSiteInfo)
         {
-            if (DependenciesBag is object)
```

Comment:
Why is it safe to not check DependenciesBag here, but we need to in the other AddDependencies methods?